src/arithmetic.c

```diff
@@ -0,0 +1,48 @@
+#include <stdint.h>
+#include "arithmetic.h"
+#include "debug.h"
+
+int add(int32_t x, int32_t y)
+{
+    DEBUG_PRINT("%d + %d", x, y);
+    return x + y;
+}
+
+int subtract(int32_t x, int32_t y)
+{
+    return x - y;
+}
+
+int multiply(int32_t x, int32_t y)
+{
+    return x * y;
+}
+
+int divide(int32_t x, int32_t y)
+{
+    if (0 == y)                 /** Prevent division by zero */
+        {
+            return -1;
```

Comment:
Use integer return code and pointer to integer to store result. Currently no way to tell the difference between:
-5 / 5 == -1 (correct)
-5 / 0 == -1 (incorrect)